### PR TITLE
Add namespaced aliases

### PR DESCRIPTION
**Pull request by zedentox**
Fix issue #455

### CODE REVIEW

**Comment by barryvdh** [comment]
Thanks! I'd rather not put the logic inside the template, can you move that to Generator class?

**Comment by zedentox** [comment]
I had a big problem with that... How to name theses variables ? 🤣

**Comment by pinepain** [comment]
Thanks for the fix! Is there anything I can help with to make this be good enough to appear in master and be tagged?

**Comment by zedentox** [comment]
@pinepain Can you give a try to this newest version ?

File: resources/views/helper.php [hunk 1]
```diff
@@ -11,16 +11,15 @@
...
-namespace <?= $namespace == '__root' ? '' : trim($namespace, '\\') ?> {
+namespace <?= $namespace == '__root' ? '' : trim($namespace, '\\') ?> { 
 <?php foreach($aliases as $alias): ?>
 
     <?= $alias->getClassType() ?> <?= $alias->getExtendsClass() ?> {
         <?php foreach($alias->getMethods() as $method): ?>
 
-        <?= trim($method->getDocComment('        ')) ?>
-
+        <?= trim($method->getDocComment('        ')) ?>
```
Comment (pinepain) [inline resources/views/helper.php]: Here is an extra whitespace after closing tag - `?> `. Minor but creates unnecessary diff.

File: resources/views/helper.php [hunk 2]
```diff
@@ -11,16 +11,15 @@
     exit("This file should not be included, only analyzed by your IDE");
 }
 
-<?php foreach($namespaces as $namespace => $aliases): ?>
+<?php foreach($namespaces_by_extends_ns as $namespace => $aliases): ?>
 <?php if ($namespace == '\Illuminate\Database\Eloquent'): continue; endif; ?>
-namespace <?= $namespace == '__root' ? '' : trim($namespace, '\\') ?> {
+namespace <?= $namespace == '__root' ? '' : trim($namespace, '\\') ?> {
```
Comment (pinepain) [inline resources/views/helper.php]: Extra whitespace at the end of line

File: resources/views/helper.php (hunk 1 above)
Comment (pinepain) [inline resources/views/helper.php]: Same below. Won't comment on every line. Unless it's intentional - give it a look, maybe it's due to some local ide settings. Apart of that - it works! Yay!

**Comment by pinepain** [comment]
@zedentox Tried a new version and it seems to be working. There is minor comment on trailing white spaces in `views/helper.php`, but apart of that - 👍. Thank you for your time and efforts!

File: resources/views/helper.php (hunk 1 above)
Comment (zedentox) [inline resources/views/helper.php]: They are intentional. If you remove some trailing whitespaces you will see that sometimes line breaks are ignored... Do you know why ?

**Comment by barryvdh** [comment]
Okay merged, let me know if it's good or not

**Comment by pinepain** [comment]
Thank you! Will do more intensive testing during this week, but the main issue looks solved.

**Comment by watercayman** [comment]
Hi folks,

I'm not at a coding level to know if I'm on the right page or not, but I wanted to send you a little more info.   I've had a similar problem to Pinepain in #455 : I can not click through to methods in my namespaced aliases -- IDE doesn't recognize them, instead clicking points into the ide_helper file.  v2.2.3 **does** successfully work, latest version, including  b611cb5 **does not**.

I register these aliases in config.app only through the aliases array like so:

```
'aliases' => [
    'CacheHelper'=> \App\Helpers\CacheHelper::class,
]
```
Code functions fine, and always has (though I'm sure I'm screwing something up), and IDE Helper has always worked perfectly in identifying these until after 2.2.3.

**Comment by watercayman** [comment]
@zedentox my apologies.  I thought this was related to this PR.  I thought the issue was namespaced aliases were not being picked up, and this PR was meant to fix that.  I was reporting that it does not for those aliases that I have registered.  Again, my apologies, I may not understand enough of this to know the difference between this and the issue looking to be fixed, though I thought they were the same.
